### PR TITLE
fix: cascader error color

### LINKED ISSUE #20343: bug: Cascader hover error when "has error"
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/form-cn/](https://ant.design/components/form-cn/)

### Steps to reproduce
https://ant.design/components/form-cn/
- 触发 form 中 Cascader 校验
- 鼠标悬浮 Cascader
- Cascader 处于 非 active 状态

### What is expected?
鼠标悬浮 Cascader 显示 红色

### What is actually happening?
鼠标悬浮显示蓝色

| Environment | Info |
|---|---|
| antd | 3.26.3 |
| React | The latest |
| System | mac win10 |
| Browser | chrome 79 |

---

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #20343

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Cascader component form check color trigger issue          |
| 🇨🇳 Chinese | 修复 Cascader 组件 Form 校验触发时，鼠标悬浮显示颜色问题          |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/form/style/index.less [hunk 1]
```diff
@@ -531,6 +531,12 @@ form {
     .active(@error-color);
   }
 
+  .@{ant-prefix}-cascader-picker {
+    .@{ant-prefix}-input {
+      border-color: @error-color;
+    }
+  }
```
Comment (afc163) [inline components/form/style/index.less]:
```less
  .@{ant-prefix}-cascader-picker {
    &:focus .@{ant-prefix}-cascader-input {
      .active(@error-color);
    }
    &:hover .@{ant-prefix}-cascader-input {
      border-color: @error-color;
    }
  }
```

可以和上面的样式合并成这样。

Comment (afc163) [reply components/form/style/index.less]: 另外，warning cascader 的样式也可以一样处理一下。

**Comment by afc163** [comment]
image

自检一下没问题就都勾上。

File: components/form/style/index.less (hunk 1 above)
Comment (xrkffgg) [reply components/form/style/index.less]: ok

**Pull request merged**